### PR TITLE
ref(convert): Notification settings by type to RTL

### PR DESCRIPTION
**Pull request by priscilawebdev**
one more day, one more conversion. This time RTL only as the file was already in Typescript :)

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx [hunk 1]
```diff
@@ -34,97 +34,81 @@ const addMockResponses = (
     method: 'GET',
     body: [],
   });
-};
+}
 
-const createWrapper = (
+function TestComponent(
```
Comment (evanpurkhiser) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:39]: I've been calling this `renderComponent`. `TestComponent` with the capitalization implies this is a react component, not a plain function

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx (hunk 1 above)
Comment (priscilawebdev) [reply static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:39]: yes it makes sense. will update it :) we will have to do a cleanup later, I guess

Comment (priscilawebdev) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:39]:
```suggestion
function renderComponent(
```

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx [hunk 2]
```diff
@@ -34,97 +34,81 @@ const addMockResponses = (
...
-};
+  render(<NotificationSettingsByType notificationType="alerts" organizations={[org]} />, {
+    context: routerContext,
+  });
+}
 
 describe('NotificationSettingsByType', function () {
   it('should render when everything is disabled', function () {
-    const wrapper = createWrapper({
+    TestComponent({
```
Comment (priscilawebdev) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:55]:
```suggestion
    renderComponent({
```

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx [hunk 3]
```diff
@@ -34,97 +34,81 @@ const addMockResponses = (
...
-    expect(fields).toHaveLength(1);
-    expect(fields.at(0).find('FieldLabel').text()).toEqual('Issue Alerts');
-    expect(fields.at(0).find('Select').text()).toEqual('Off');
+    expect(screen.getByRole('textbox', {name: 'Issue Alerts'})).toBeInTheDocument();
+    expect(screen.getByText('Off')).toBeInTheDocument();
   });
 
   it('should render when notification settings are enabled', function () {
-    const wrapper = createWrapper({
+    TestComponent({
```
Comment (priscilawebdev) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:65]:
```suggestion
    renderComponent({
```

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx [hunk 4]
```diff
@@ -34,97 +34,81 @@ const addMockResponses = (
...
+    expect(screen.getByRole('textbox', {name: 'Delivery Method'})).toBeInTheDocument();
+    expect(screen.getByText('Email')).toBeInTheDocument(); // Select Value
+    expect(screen.getByText('Slack')).toBeInTheDocument(); // Select Value
   });
 
   it('should render warning modal when identity not linked', function () {
     const org = TestStubs.Organization();
-    const wrapper = createWrapper(
+
+    TestComponent(
```
Comment (priscilawebdev) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:80]:
```suggestion
    renderComponent(
```

File: static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx [hunk 5]
```diff
@@ -34,97 +34,81 @@ const addMockResponses = (
...
+    ).toBeInTheDocument();
+
+    expect(screen.getByRole('listitem')).toHaveTextContent(org.slug);
   });
 
   it('should not render warning modal when identity is linked', function () {
     const org = TestStubs.Organization();
-    const wrapper = createWrapper(
+
+    TestComponent(
```
Comment (priscilawebdev) [inline static/app/views/settings/account/notifications/notificationSettingsByType.spec.tsx:100]:
```suggestion
    renderComponent(
```

**Pull request merged**